packages/mui-base/src/utils/useExtendedForkRef.ts

```diff
@@ -0,0 +1,14 @@
+import type { BaseUIComponentProps } from './BaseUI.types';
+import { useForkRef } from './useForkRef';
+
+/**
+ * Merges the rendering element's `ref` in addition to the other `ref`s.
+ * @ignore - internal hook.
+ */
+export function useExtendedForkRef<ElementType extends React.ElementType, OwnerState>(
```

Comment:
This name doesn't tell much about the purpose of this function. How about `useRenderPropForkedRef`?
I do realize that `forkRef` is not the best name either, though, so perhaps we can use another term, like `useRenderPropSyncedRef`?

Reply:
Not sure I like any name, I tried `useForkRefWithRenderProp`